### PR TITLE
IBX-5108: X-Expected-User

### PR DESCRIPTION
**Pull request by adriendupuis**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-5108](https://issues.ibexa.co/browse/IBX-5108)
| Versions      | 4.5.1, 4.6
| Edition       | All

Documents the `X-Expected-User` REST request header

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale (header list and added section only)
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/api/rest_api/rest_api_usage/rest_requests.md [hunk 1]
```diff
@@ -136,6 +137,14 @@ Examples of such requests are:
 - [moving a Location and its subtree](../rest_api_reference/rest_api_reference.html#managing-content-move-subtree)
 - [swapping a Location with another](../rest_api_reference/rest_api_reference.html#managing-content-swap-location)
 
+### Expected user
+
+The `X-Expected-User` header specifies the user needed on client side for the request execution.
```
Comment (Steveb-p) [inline docs/api/rest_api/rest_api_usage/rest_requests.md:142]: This part of the sentence does not convey any information and might be confusing.
```suggestion
The `X-Expected-User` header specifies the user needed for the request execution.
```

File: docs/api/rest_api/rest_api_usage/rest_requests.md [hunk 2]
```diff
@@ -136,6 +137,14 @@ Examples of such requests are:
...
+If the current username on server side isn't equal to `X-Expected-User` value, a `401 Unauthorized` error is returned.
```
Comment (Steveb-p) [inline docs/api/rest_api/rest_api_usage/rest_requests.md:143]: A sentence explaining behavior without the `X-Expected-User` header would in my opinion be helpful here.

Defaulting to anonymous user is implied in the next paragraph, but it isn't explicitly said.

**Review by Steveb-p — APPROVED** [review]

File: docs/api/rest_api/rest_api_usage/rest_requests.md (hunk 2 above)
Comment (adriendupuis) [reply docs/api/rest_api/rest_api_usage/rest_requests.md:143]: @Steveb-p What about this?:
```suggestion
With this header, if the current username on server side isn't equal to `X-Expected-User` value, a `401 Unauthorized` error is returned.
Without this header, the request will be executed with the current user who might be unexpected (like the Anonymous user if a previous authentication has expired) and an ambiguous response might be returned without a clue that the request was a success but with a wrong user.
```

Comment (Steveb-p) [reply docs/api/rest_api/rest_api_usage/rest_requests.md:143]: Sounds good to me :+1:

File: docs/api/rest_api/rest_api_usage/rest_requests.md [hunk 3]
```diff
@@ -136,6 +137,15 @@ Examples of such requests are:
 - [moving a Location and its subtree](../rest_api_reference/rest_api_reference.html#managing-content-move-subtree)
 - [swapping a Location with another](../rest_api_reference/rest_api_reference.html#managing-content-swap-location)
 
+### Expected user
+
+The `X-Expected-User` header specifies the user needed for the request execution.
+With this header, if the current username on server side isn't equal to `X-Expected-User` value, a `401 Unauthorized` error is returned.
+Without this header, the request is executed with the current user who might be unexpected (like the Anonymous user if a previous authentication has expired) and an ambiguous response might be returned without a clue that the request was a success but with a wrong user.
```
Comment (MagdalenaZuba) [inline docs/api/rest_api/rest_api_usage/rest_requests.md:144]:
```suggestion
Without this header, the request is executed with the current user who might be unexpected (like the Anonymous user if a previous authentication has expired) and an ambiguous response might be returned without an information that the request was a success but with a wrong user.
```

**Review by MagdalenaZuba — APPROVED** [review]

File: docs/api/rest_api/rest_api_usage/rest_requests.md (hunk 3 above)
Comment (adriendupuis) [reply docs/api/rest_api/rest_api_usage/rest_requests.md:144]: @MagdalenaZuba Maybe shorter?:
```suggestion
Without this header, the request is executed with the current user who might be unexpected (like the Anonymous user if a previous authentication has expired) and an ambiguous response might be returned as a success not telling with a wrong user.
```

**Pull request merged**